File: kotlin_obdii/app/src/test/java/com/rheosoft/obdii/screenmodels/OnboardingScreenModelTest.kt
package com.rheosoft.obdii.screenmodels

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertNull
import kotlin.test.assertTrue

class OnboardingScreenModelTest {
    @Test
    fun `has nine intro pages ending with demo`() {
        assertEquals(9, OnboardingScreenModel.pages.size)
        assertEquals(OnboardingPageKind.Demo, OnboardingScreenModel.pages.last().kind)
    }

    @Test
    fun `demo page is last and highlights gauges tab`() {
        assertTrue(OnboardingScreenModel.isDemoPage(OnboardingScreenModel.DEMO_PAGE_INDEX))
        assertEquals(
            OnboardingScreenModel.GAUGES_TAB_INDEX,
            OnboardingScreenModel.previewTabIndex(OnboardingScreenModel.DEMO_PAGE_INDEX),
        )
    }

    @Test
    fun `gauge picker page opens picker and hides nav highlight`() {
        val idx = OnboardingScreenModel.GAUGE_PICKER_PAGE_INDEX
        assertTrue(OnboardingScreenModel.showGaugePicker(idx))
        assertTrue(OnboardingScreenModel.usesCompactScrim(idx))
        assertNull(OnboardingScreenModel.highlightedNavTab(idx))
    }

    @Test
    fun `gauges dashboard page uses compact scrim`() {
        val idx = OnboardingScreenModel.pages.indexOfFirst {
            it.kind == OnboardingPageKind.TabTour && it.previewTabIndex == OnboardingScreenModel.GAUGES_TAB_INDEX
        }
        assertTrue(OnboardingScreenModel.isGaugesDashboardPage(idx))
        assertTrue(OnboardingScreenModel.usesCompactScrim(idx))
        assertEquals(OnboardingScreenModel.GAUGES_TAB_INDEX, OnboardingScreenModel.highlightedNavTab(idx))
    }

    @Test
    fun `welcome has no live tab preview and shows summary`() {
        assertNull(OnboardingScreenModel.previewTabIndex(0))
        assertTrue(OnboardingScreenModel.showWelcomeSummary(0))
        assertEquals(5, OnboardingScreenModel.welcomeSummaryPoints.size)
    }

    @Test
    fun `last page detection`() {
        assertFalse(OnboardingScreenModel.isLastPage(0))
        assertTrue(OnboardingScreenModel.isLastPage(OnboardingScreenModel.pages.lastIndex))
    }
}

File: flutter_obdii/test/screenmodels/onboarding_screen_model_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:flutter_obdii/screenmodels/onboarding_screen_model.dart';

void main() {
  test('has nine intro pages ending with demo', () {
    expect(OnboardingScreenModel.pages.length, 9);
    expect(
      OnboardingScreenModel.pages.last.kind,
      OnboardingPageKind.demo,
    );
  });

  test('demo page is last and highlights gauges tab', () {
    expect(
      OnboardingScreenModel.isDemoPage(OnboardingScreenModel.demoPageIndex),
      isTrue,
    );
    expect(
      OnboardingScreenModel.previewTabIndex(OnboardingScreenModel.demoPageIndex),
      OnboardingScreenModel.gaugesTabIndex,
    );
  });

  test('gauge picker page opens picker and hides nav highlight', () {
    final idx = OnboardingScreenModel.gaugePickerPageIndex;
    expect(OnboardingScreenModel.showGaugePicker(idx), isTrue);
    expect(OnboardingScreenModel.usesCompactScrim(idx), isTrue);
    expect(OnboardingScreenModel.highlightedNavTab(idx), isNull);
  });

  test('gauges dashboard page uses compact scrim', () {
    final idx = OnboardingScreenModel.pages.indexWhere(
      (p) =>
          p.kind == OnboardingPageKind.tabTour &&
          p.previewTabIndex == OnboardingScreenModel.gaugesTabIndex,
    );
    expect(OnboardingScreenModel.isGaugesDashboardPage(idx), isTrue);
    expect(OnboardingScreenModel.usesCompactScrim(idx), isTrue);
    expect(
      OnboardingScreenModel.highlightedNavTab(idx),
      OnboardingScreenModel.gaugesTabIndex,
    );
  });

  test('welcome has no live tab preview and shows summary', () {
    expect(OnboardingScreenModel.previewTabIndex(0), isNull);
    expect(OnboardingScreenModel.showWelcomeSummary(0), isTrue);
    expect(OnboardingScreenModel.welcomeSummaryPoints.length, 5);
  });

  test('last page detection', () {
    expect(OnboardingScreenModel.isLastPage(0), isFalse);
    expect(
      OnboardingScreenModel.isLastPage(OnboardingScreenModel.pages.length - 1),
      isTrue,
    );
  });
}
